Sources/ArcGISToolkit/Components/Augmented Reality/CalibrationView.swift

```diff
@@ -0,0 +1,193 @@
+                                .foregroundStyle(.secondary)
+                            Spacer()
+                            Text((viewModel.calibrationElevation ?? viewModel.cameraController.originCamera.location.z ?? 0), format: .number.precision(.fractionLength(0)))
+                            Spacer()
+                        }
+                    } onIncrement: {
+                        updateElevation(by: 1)
+                    } onDecrement: {
+                        updateElevation(by: -1)
+                    }
+                }
+                JoystickSliderView()
+                    .onSliderValueChanged { delta in
+                        updateElevation(by: delta)
+                    }
+            }
+        }
+        
+        @ViewBuilder
+        var dismissButton: some View {
+            Button {
+                withAnimation {
+                    viewModel.isCalibrating = false
+                }
+            } label: {
+                Image(systemName: "xmark.circle.fill")
+                    .resizable()
+                    .symbolRenderingMode(.hierarchical)
+                    .foregroundStyle(.secondary)
+                    .frame(width: 28, height: 28)
+            }
+            .buttonStyle(.plain)
+        }
+        
+        /// Rotates the heading of the scene view camera controller by the heading delta in degrees.
+        /// - Parameter headingDelta: The heading delta in degrees.
+        private func rotateHeading(by headingDelta: Double) {
+            let originCamera = viewModel.cameraController.originCamera
+            let newHeading = originCamera.heading + headingDelta
+            viewModel.cameraController.originCamera = originCamera.rotatedTo(
+                heading: newHeading,
+                pitch: originCamera.pitch,
+                roll: originCamera.roll
+            )
+            viewModel.calibrationHeading = newHeading
+        }
+        
+        /// Elevates the scene view camera controller by the elevation delta.
+        /// - Parameter elevationDelta: The elevation delta.
+        private func updateElevation(by elevationDelta: Double) {
+            viewModel.cameraController.originCamera = viewModel.cameraController.originCamera.elevated(by: elevationDelta)
+            if let elevation = viewModel.cameraController.originCamera.location.z {
+                viewModel.calibrationElevation = elevation
+            }
+        }
+    }
+}
+
+/// A view for a joystick style slider.
+struct JoystickSliderView: View {
+    /// The slider value.
+    @State private var value = 0.0
+    /// The timer for the "joystick" behavior.
+    @State private var timer: Timer?
+    /// The delta amount based on the slider value.
+    private var joystickDelta: Double {
+        Double(signOf: value, magnitudeOf: value * value / 25)
+    }
+    /// User defined action to be performed when the slider value changes.
+    var sliderValueChangedAction: ((Double) -> Void)? = nil
```

Comment:
is this the slider value changing, or the slider delta changing? It seems to be the delta in this case correct? If so then name and comment should reflect that.

Reply:
Yes the slider delta value is changing, the names have been updated.